### PR TITLE
✨ Add StepSecurity to docs

### PR DESCRIPTION
**Pull request by laurentsimon**
[StepSecurity's online tool](https://app.stepsecurity.io/) helps users determine:
1. the permission to use for they workflows
2. the hash for an action, if it currently uses a version

Both are extremely useful to drive changes along with the scorecard action.
The tool is open-source and free.

### CODE REVIEW

File: docs/checks.md
Comment (laurentsimon) [inline docs/checks.md:60]: please ignore this file: some changes are due to a previous PR that did not keep the .yaml and .md in sync after updates.

**Review by naveensrinivasan — APPROVED** [review]
Thanks! Based on the offline discussion we have agreed on the step security tool because it is OSS and I am fine we recommending the tool that is OSS.

**Comment by laurentsimon** [comment]
> Thanks! Based on the offline discussion we have agreed on the step security tool because it is OSS and I am fine we recommending the tool that is OSS.

agreed.

**Comment by laurentsimon** [comment]
cc @varunsh-coder Thanks for the hard work!

**Comment by varunsh-coder** [comment]
Thanks a lot @laurentsimon! Happy to see this PR. https://github.com/step-security/secure-workflows will stay open source

**Comment by evverx** [comment]
> The tool is open-source and free.

@laurentsimon looking at https://github.com/step-security/secure-workflows/blob/main/LICENSE it seems it isn't open source. It was actually relicenced 20 days ago: https://github.com/step-security/secure-workflows/commit/94494275e57a536c38956dfac8a801dd738bb660#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7

**Comment by varunsh-coder** [comment]
> > The tool is open-source and free.
> 
> @laurentsimon looking at https://github.com/step-security/secure-workflows/blob/main/LICENSE it seems it isn't open source. It was actually relicenced 20 days ago: [step-security/secure-workflows@9449427#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7](https://github.com/step-security/secure-workflows/commit/94494275e57a536c38956dfac8a801dd738bb660#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7)

@evverx I didn't understand. I used the same license as [Dependabot](https://github.com/dependabot/dependabot-core). I believe this means it is free for non-commercial use. Please let me know specific concern with the license, and I can update it accordingly...

**Comment by evverx** [comment]
> I believe this means it is free for non-commercial use

@varunsh-coder at the time this PR was opened it was free for commercial use as well and I suspect this PR wouldn't have been merged if it had been known that it would change in a week.

**Comment by evverx** [comment]
Just to clarify I ran into this license when the knowledge database was discussed in https://github.com/google/clusterfuzzlite/issues/86#issuecomment-1032119662 and I thought since scorecard recommends it I was actually discussing code that can actually be incorporated into other open-source projects but that license prohibits that obviously because it can't be integrated into projects that can be used by everybody.

**Comment by evverx** [comment]
Before I go, I'm not trying bring the licence back (I'll never use StepSecurity). It's just that it would have been nice if in that issue it had been made clear upfront that the knowledge database is essentially a commercial product and I just wouldn't have participated in that discussion.

**Comment by naveensrinivasan** [comment]
> > I believe this means it is free for non-commercial use
> 
> 
> 
> @varunsh-coder at the time this PR was opened it was free for commercial use as well and I suspect this PR wouldn't have been merged if it had been known that it would change in a week.

I agree. If it was the case I wouldn't have recommended. 

I am surprised it was changed after it was merged in. 

It was agreed verbally that we wouldn't recommend any commercial tool within Scorecard.

**Comment by naveensrinivasan** [comment]
We need to discuss this.

**Comment by varunsh-coder** [comment]
> We need to discuss this.

@naveensrinivasan sure, we can discuss it. I am not fixed on any license. I was not part of the discussions, and don't really know the requirements. I used the same license as Dependabot, and that is also listed as a way to remediate some checks. Also, step-security/secure-workflows is not commercial, no one has to pay to use it.

**Comment by evverx** [comment]
Dependabot, LGTM and so on are GitHub products and they have never been hidden behind open source licenses. To be honest I'm not sure what's to discuss here but luckily I'm not a scorecard maintainer :-)

**Comment by evverx** [comment]
@laurentsimon @naveensrinivasan I'd be curious to find out what was decided in the end. Personally I don't think scorecard should promote codebases switching licenses like that but it's obviously not my call. I took a look at the last biweekly meeting and as far as I can tell it wasn't discussed there (publicly at least).

**Comment by naveensrinivasan** [comment]
> @laurentsimon @naveensrinivasan I'd be curious to find out what was decided in the end. Personally I don't think scorecard should promote codebases switching licenses like that but it's obviously not my call. I took a look at the last biweekly meeting and as far as I can tell it wasn't discussed there (publicly at least).

@evverx  We didn't get a chance to discuss.

**Comment by varunsh-coder** [comment]
I have reverted the license until this discussion is resolved. If I had known it would cause confusion/ concern, I would have had the discussion before changing the license.

**Comment by evverx** [comment]
The site still keeps recommending "harden-runner" by default, which uses https://github.com/step-security/agent under the hood. The agent was relicensed in https://github.com/step-security/agent/commit/e541b50b6135796d9c4d41b59ad1180e0a22cb02 (and hasn't been relicensed back) and I assume that's the "patent-pending" technology the site briefly mentions.

Anyway, @laurentsimon @naveensrinivasan @azeemshaikh38 given that it's been several days I wonder if it is safe to assume that `scorecard` is going to keep recommending that?

**Comment by naveensrinivasan** [comment]
@david-a-wheeler Would you be able to provide some input on this?

**Comment by azeemshaikh38** [comment]
So here's my take on this - Dependabot is something recommended by GitHub (or the VCS platform) itself (see this [link](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates)). So Scorecard would recommend Dependabot irrespective of its license and we can't really compare Step-Security to Dependabot in that sense.

A more fair comparison to me seems like RenovateBot. And AFAICT, RenovateBot is free to use commercially or otherwise (see [license](https://github.com/renovatebot/renovate/blob/main/license)). I think we jumped the gun with Step-Security by not explicitly agreeing on the license requirements of Scorecard/OpenSSF recommendations.

So this is what I would suggest - @varunsh-coder since you mentioned that you are not fixed on the license, could you please use a license similar to RenovateBot (i.e free to use for anyone)? Meanwhile, it's a larger discussion we as Scorecard/OpenSSF need to have about what restrictions we put on tools recommended by our community.

**Comment by varunsh-coder** [comment]
> So here's my take on this - Dependabot is something recommended by GitHub (or the VCS platform) itself (see this [link](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates)). So Scorecard would recommend Dependabot irrespective of its license and we can't really compare Step-Security to Dependabot in that sense.
> 
> A more fair comparison to me seems like RenovateBot. And AFAICT, RenovateBot is free to use commercially or otherwise (see [license](https://github.com/renovatebot/renovate/blob/main/license)). I think we jumped the gun with Step-Security by not explicitly agreeing on the license requirements of Scorecard/OpenSSF recommendations.
> 
> So this is what I would suggest - @varunsh-coder since you mentioned that you are not fixed on the license, could you please use a license similar to RenovateBot (i.e free to use for anyone)? Meanwhile, it's a larger discussion we as Scorecard/OpenSSF need to have about what restrictions we put on tools recommended by our community.

Thanks @azeemshaikh38 for the response. Let me read more about `GNU Affero General Public License` which is used by renovatebot. I should be able to get back on that in a couple of days. 

I also wanted to get clarification on other tools/ services StepSecurity is building that are not directly recommended by scorecard, but that users may discover as a result of going to https://app.stepsecurity.io. Does all of that need to be in a public repo and need to use one of the allowed licenses? or does this discussion apply specifically to https://github.com/step-security/secure-workflows? 

Overall, as you said, having more clarity on licenses that tools should use as a requirement to be recommended would be helpful. This thread has been uncomfortable for me and I suspect it will be uncomfortable for others who may want to write tools in this space, unless the requirements are very clear.

**Comment by evverx** [comment]
> it's a larger discussion we as Scorecard/OpenSSF need to have about what restrictions we put on tools recommended by our community

I'm not part of that community but just in case it's helpful personally I have absolute no problem with companies selling/promoting their products as long as they're upfront about that. The fact that open-source licenses are used for gathering feedback from developers, marketing or whatever and then are changed on the fly is what I'm concerned about.

> it will be uncomfortable for others who may want to write tools in this space

I don't think it will. Vendors are usually upfront about what they do. It's not uncommon for startups/companies to reach out to open source projects and offer their services for free (which usually let them use those projects to help to sell their products or just promote them) and for example once those startups are sold the services are cut down but then again it's transparent and is discussed in the beginning and nobody is surprised when that happens.

**Pull request merged**
